---
title: "Divergent Symbiodiniaceae Thermal Tolerances"
author: "Alexa Huzar, Jacob Jaskiel, Yaoyuan Gan"
date: "3/24/2021"
output: html_document
---

# Introduction

Different Symbiodiniaceae species and populations can convey different thermal tolerances to their coral host. However, the genes that control symbiont thermal tolerance are still unclear. Levin et al. (2016) attempted to address this gap in knowledge by comparing two heterogeneous populations of type C1 *Symbiodinium* from the same species of coral host (*Acropora tenuis*) living in two distinct sites on the Great Barrier Reef. The population from South Molle Island (SM) is considered thermally sensitive while the population from Magnetic Island (MI) is thermally tolerant allowing them to compare differentially expressed genes based on thermal tolerance level. We are now using their data but analyzing it through a different pathway to determine what genes are differentially expressed as a result of temperature stress.

**R version** 

We used R version R-4.0.3 ADD in package versions before submitting

# DESeq2 and GO enrichment pathways
In order to understand the genes that are differentially expressed due to thermal tolerance, we used the Deseq2 and Go enrichment pathways. Deseq2 allows us to compare gene counts between treatment to determine what is differentially expressed. This data is then funneled into the Go enrichment pathway to determine the functions of the differentially expressed genes.

# Deseq2 pipeline

### Preparation Steps
**We first loaded the appropriate libraries for all the analysis and graphs.**

```{r, loadlib, echo=TRUE, results='hide', message=FALSE, warning=FALSE}
library(DESeq2) 
library(affycoretools)
library(arrayQualityMetrics)
library(genefilter)
library(Biobase)
library(ggplot2)
library(dplyr)
library(tidyverse)
library(pheatmap)
library(vegan)
library(ggrepel)
```

Set your working directory
```{r}
setwd("C:/Users/Bruce Gan/Desktop/BI586/HW2")
```
Now we read in our count data
```{r} 
countData <- read.table("LevinCounts.txt")
head(countData)
length(countData[,1])
#### length = 31114
```

**Now we are changing the names of our columns to make it easier to read and understand. The first part of the name refers to the site the sample was from and the second part refers to the temperature the sample was taken at. All samples were taken after 13 days at their respective temperatures. 
```{r}
names(countData)=c( "MI_32a", "MI_32b", "MI_27a", "MI_27b", "SM_32a", "SM_32b", "SM_27a", "SM_27b")
row.names(countData)=sub("", "isogroup", rownames(countData))
head(countData)
```


###-------Not sure we need this step now. I think its just for the Go enrichment
#Make sure you can succesfully read in the iso2gene tab separated file
gg=read.table("Crep454_iso2gene.tab",sep="\t",quote="", row.names=1) 

### Check for outliers

First set the working directory and create a condtions table
```{r}
setwd("C:/Users/Bruce Gan/Desktop/BI586/HW2/outlier")
v=setwd("C:/Users/Bruce Gan/Desktop/BI586/HW2/outlier")
treat=c( "MI_32a", "MI_32b", "MI_27a", "MI_27b", "SM_32a", "SM_32b", "SM_27a", "SM_27b")
g=data.frame(treat)
g
colData= g
```

```{r}
countData <- round(countData, digits=0)
```

**DeSeq now creates our model for us based on how does gene expression vary by treatment**
```{r}
dds=DESeqDataSetFromMatrix(countData=countData,
                           colData = g,
                           design = ~treat)

vsd.ge=assay(vst(dds))
rl=vst(dds)
e=ExpressionSet(assay(rl), AnnotatedDataFrame(as.data.frame(colData(rl))))
arrayQualityMetrics(e,outdir=v,intgroup=c("treat"),force=T)
```

**No outliers found!!**

#Going back to main working directory
```{r}
setwd("C:/Users/Bruce Gan/Desktop/BI586/HW2")
```

#read in counts Same stuff again after outliers
```{r}
countData <- read.table("LevinCounts.txt")
head(countData)
length(countData[,1])
```
#31114

```{r}
names(countData)=c( "MI_32a", "MI_32b", "MI_27a", "MI_27b", "SM_32a", "SM_32b", "SM_27a", "SM_27b")
row.names(countData)=sub("", "isogroup", rownames(countData))
head(countData)
```

##total number of raw counts. Useful to report
```{r}
totalCounts=colSums(countData)
totalCounts
barplot(totalCounts, col=c("coral", "coral", "coral", "red", "red", "red", "blue", "blue"), ylab="raw counts")
```

**MI_32a  MI_32b  MI_27a  MI_27b  SM_32a  SM_32b  SM_27a  SM_27b**
**2968600 3160913 3025741 3014954 3840001 3629641 2930790 3279507**


creating call data again
```{r}
treat=c( "MI_32", "MI_32", "MI_27", "MI_27", "SM_32", "SM_32", "SM_27", "SM_27")
g=data.frame(treat)
g
colData= g
countData <- round(countData, digits=0)
```

###same deseq object as above
```{r}
dds=DESeqDataSetFromMatrix(countData=countData,
                           colData = g,
                           design = ~treat)
```

#one step DESeq. Can run ones that are way more complicated 
```{r}
dds<-DESeq(dds)
```
# estimating size factors
# estimating dispersions
# gene-wise dispersion estimates
# mean-dispersion relationship
# final dispersion estimates
# fitting model and testing

```{r}
keep <- rowSums(counts(dds) >= 10) >= 3
dds<-dds[keep,]
head(dds)
res<- results(dds)
```

#Look at dispersions plot. We want it to look like a hockey stick
###normalization method. Shows what DEseq2 is doing to your data
```{r}
plotDispEsts(dds, main="Dispersion plot Symbionts")
```

#####MI_27 vs MI_32 pairwise comparisons. Starting to look for DE genes
#treatment first control second
```{r}
colData$MI32<-factor(colData$treat, levels=c("MI_32","MI_27")) 
resMI32 <- results(dds, contrast=c("treat","MI_32","MI_27"))
```
#how many FDR < 10%?
```{r}
table(resMI32$padj<0.01)
```
# 0.1=4248
# 0.05=3133
# 0.01=1646

###p adjusted value accounts for comparsions across different treatments.Account for all the times you are comparing things

##shows genes removed and totals up and downregulated relative to control
```{r}
summary(resMI32)
```

##another way to look at it 
```{r}
nrow(resMI32[resMI32$padj<0.01 & !is.na(resMI32$padj),]) 
```
# Num significantly differentially expressed genes excluding the no/low count genes   

dev.off()
### mean of normalized counts. far right is more highly expressed genes, 
###more likely to find DE genes a=among those highly expressed. hard to find when low coverage
```{r}
plotMA(resMI32, main="MI_32 vs MI_27")
```

```{r}
plotMA(resMI32, main="MI_32 vs MI_27", ylim=c(-20,20))
```

##put results in dataframe
```{r}
results <- as.data.frame(resMI32)
head(results)
```
##another way to look at up and down regulated genes
```{r}
nrow(resMI32[resMI32$padj<0.1 & resMI32$log2FoldChange > 0 & !is.na(resMI32$padj),])
nrow(resMI32[resMI32$padj<0.1 & resMI32$log2FoldChange < 0 & !is.na(resMI32$padj),])
```
**UP in MI32 2087**
**DOWN in Mi32 2161**

###write data in a table
```{r}
write.table(resMI32, file="MI_32.txt", quote=F, sep="\t")
```
###read the table back in
```{r}
cd <- read.table("MI_32.txt")
head(cd)
```


###this is taking each isogroup and making ranked p value with directionality
```{r}
cd
go_input_MI32 = cd %>%
  tibble::rownames_to_column(var = "iso") %>%
  mutate(mutated_p = -log(pvalue)) %>%
  mutate(mutated_p_updown = ifelse(log2FoldChange < 0, mutated_p*-1, mutated_p*1)) %>%
  na.omit() %>%
  select(iso, mutated_p_updown)
head(go_input_MI32)
colnames(go_input_MI32) <- c("gene", "pval")
head(go_input_MI32)
write.csv(go_input_MI32, file="MI32_GO.csv", quote=F, row.names=FALSE)
```

### SM_32 versus SM_27

```{r}
colData$SM32<-factor(colData$treat, levels=c("SM_32","SM_27")) 
resSM32 <- results(dds, contrast=c("treat","SM_32","SM_27"))
```
#how many FDR < 10%?
```{r}
table(resSM32$padj<0.01)
```
# 0.1=6701
# 0.05=5259
# 0.01=3342

###p adjusted value accounts for comparsions across different treatments.Account for all the times you are comparing things

##shows genes removed and totals up and downregulated relative to control
```{r}
summary(resSM32)
```

##another way to look at it 
```{r}
nrow(resSM32[resSM32$padj<0.01 & !is.na(resSM32$padj),]) 
```
# Num significantly differentially expressed genes excluding the no/low count genes   

dev.off()
### mean of normalized counts. far right is more highly expressed genes, 
###more likely to find DE genes a=among those highly expressed. hard to find when low coverage
```{r}
plotMA(resSM32, main="SM_32 vs SM_27")
```

```{r}
plotMA(resSM32, main="SM_32 vs SM_27", ylim=c(-2,2))
```
##put results in dataframe
```{r}
results <- as.data.frame(resSM32)
head(results)
```
##another way to look at up and down regulated genes
```{r}
nrow(resSM32[resMI32$padj<0.1 & resSM32$log2FoldChange > 0 & !is.na(resSM32$padj),])
nrow(resSM32[resMI32$padj<0.1 & resSM32$log2FoldChange < 0 & !is.na(resSM32$padj),])
```
**UP in SM32 2541**
**DOWN in SM32 1706**

###write data in a table
```{r}
write.table(resSM32, file="SM_32.txt", quote=F, sep="\t")
```
###read the table back in
```{r}
cd <- read.table("SM_32.txt")
head(cd)
```


###this is taking each isogroup and making ranked p value with directionality
```{r}
cd
go_input_SM32 = cd %>%
  tibble::rownames_to_column(var = "iso") %>%
  mutate(mutated_p = -log(pvalue)) %>%
  mutate(mutated_p_updown = ifelse(log2FoldChange < 0, mutated_p*-1, mutated_p*1)) %>%
  na.omit() %>%
  select(iso, mutated_p_updown)
head(go_input_SM32)
colnames(go_input_SM32) <- c("gene", "pval")
head(go_input_SM32)
write.csv(go_input_SM32, file="SM32_GO.csv", quote=F, row.names=FALSE)
```

### SM_32 versus MI_32

```{r}
colData$BOTH32<-factor(colData$treat, levels=c("MI_32","SM_32")) 
resBOTH32 <- results(dds, contrast=c("treat","MI_32","SM_32"))
```
#how many FDR < 10%?
```{r}
table(resBOTH32$padj<0.01)
```
# 0.1=3704
# 0.05=2814
# 0.01=1738

###p adjusted value accounts for comparsions across different treatments.Account for all the times you are comparing things

##shows genes removed and totals up and downregulated relative to control
```{r}
summary(resBOTH32)
```

##another way to look at it 
```{r}
nrow(resBOTH32[resBOTH32$padj<0.01 & !is.na(resBOTH32$padj),]) 
```
# Num significantly differentially expressed genes excluding the no/low count genes   

dev.off()
### mean of normalized counts. far right is more highly expressed genes, 
###more likely to find DE genes a=among those highly expressed. hard to find when low coverage
```{r}
plotMA(resBOTH32, main="MI_32 vs SM_32")
```

```{r}
plotMA(resBOTH32, main="MI_32 vs SM_32", ylim=c(-20,20))
```
##put results in dataframe
```{r}
results <- as.data.frame(resBOTH32)
head(results)
```
##another way to look at up and down regulated genes
```{r}
nrow(resBOTH32[resBOTH32$padj<0.1 & resBOTH32$log2FoldChange > 0 & !is.na(resBOTH32$padj),])
nrow(resBOTH32[resBOTH32$padj<0.1 & resBOTH32$log2FoldChange < 0 & !is.na(resBOTH32$padj),])
```
**UP in BOTH32 2682**
**DOWN in BOTH32 1022**

###write data in a table
```{r}
write.table(resBOTH32, file="BOTH_32.txt", quote=F, sep="\t")
```
###read the table back in
```{r}
cd <- read.table("BOTH_32.txt")
head(cd)
```


###this is taking each isogroup and making ranked p value with directionality
```{r}
cd
go_input_BOTH32 = cd %>%
  tibble::rownames_to_column(var = "iso") %>%
  mutate(mutated_p = -log(pvalue)) %>%
  mutate(mutated_p_updown = ifelse(log2FoldChange < 0, mutated_p*-1, mutated_p*1)) %>%
  na.omit() %>%
  select(iso, mutated_p_updown)
head(go_input_BOTH32)
colnames(go_input_BOTH32) <- c("gene", "pval")
head(go_input_BOTH32)
write.csv(go_input_BOTH32, file="BOTH32_GO.csv", quote=F, row.names=FALSE)
```

### SM_27 versus MI_27

```{r}
colData$BOTH27<-factor(colData$treat, levels=c("MI_27","SM_27")) 
resBOTH27 <- results(dds, contrast=c("treat","MI_27","SM_27"))
```
#how many FDR < 10%?
```{r}
table(resBOTH27$padj<0.01)
```
# 0.1=5204
# 0.05=3991
# 0.01=2325

###p adjusted value accounts for comparsions across different treatments.Account for all the times you are comparing things

##shows genes removed and totals up and downregulated relative to control
```{r}
summary(resBOTH27)
```

##another way to look at it 
```{r}
nrow(resBOTH27[resBOTH27$padj<0.01 & !is.na(resBOTH27$padj),]) 
```
# Num significantly differentially expressed genes excluding the no/low count genes   

dev.off()
### mean of normalized counts. far right is more highly expressed genes, 
###more likely to find DE genes a=among those highly expressed. hard to find when low coverage
```{r}
plotMA(resBOTH27, main="MI_27 vs SM_27")
```

```{r}
plotMA(resBOTH32, main="MI_27 vs SM_27", ylim=c(-20,20))
```
##put results in dataframe
```{r}
results <- as.data.frame(resBOTH27)
head(results)
```
##another way to look at up and down regulated genes
```{r}
nrow(resBOTH27[resBOTH27$padj<0.1 & resBOTH27$log2FoldChange > 0 & !is.na(resBOTH27$padj),])
nrow(resBOTH27[resBOTH27$padj<0.1 & resBOTH27$log2FoldChange < 0 & !is.na(resBOTH27$padj),])
```
**UP in BOTH27 2771**
**DOWN in BOTH27 2433**

###write data in a table
```{r}
write.table(resBOTH27, file="BOTH_27.txt", quote=F, sep="\t")
```
###read the table back in
```{r}
cd <- read.table("BOTH_27.txt")
head(cd)
```


###this is taking each isogroup and making ranked p value with directionality
```{r}
cd
go_input_BOTH27 = cd %>%
  tibble::rownames_to_column(var = "iso") %>%
  mutate(mutated_p = -log(pvalue)) %>%
  mutate(mutated_p_updown = ifelse(log2FoldChange < 0, mutated_p*-1, mutated_p*1)) %>%
  na.omit() %>%
  select(iso, mutated_p_updown)
head(go_input_BOTH27)
colnames(go_input_BOTH27) <- c("gene", "pval")
head(go_input_BOTH27)
write.csv(go_input_BOTH27, file="BOTH27_GO.csv", quote=F, row.names=FALSE)
```


### Binding two columns and pulling out pvalues. NA means p value is too high
```{r}
valMI32=cbind(resMI32$pvalue, resMI32$padj)
head(valMI32)
colnames(valMI32)=c("pval.MI32", "padj.MI32")
length(valMI32[,1])
table(complete.cases(valMI32))
```
```{r}
valSM32=cbind(resSM32$pvalue, resSM32$padj)
head(valSM32)
colnames(valSM32)=c("pval.SM32", "padj.SM32")
length(valSM32[,1])
table(complete.cases(valSM32))
```
```{r}
valBOTH32=cbind(resBOTH32$pvalue, resBOTH32$padj)
head(valBOTH32)
colnames(valBOTH32)=c("pval.BOTH32", "padj.BOTH32")
length(valBOTH32[,1])
table(complete.cases(valBOTH32))
```
```{r}
valBOTH27=cbind(resBOTH27$pvalue, resBOTH27$padj)
head(valBOTH27)
colnames(valBOTH27)=c("pval.BOTH27", "padj.BOTH27")
length(valBOTH27[,1])
table(complete.cases(valBOTH27))
```

### Make rlogdata and pvals table
**Normalization method, important for heatmap and PCA**
```{r}
rlog=rlogTransformation(dds, blind=TRUE) 
rld=assay(rlog)
head(rld)
colnames(rld)=paste(colData$treat)
head(rld)
length(rld[,1])
```

#bind together pvalues and rlog data 
**FALSE  TRUE**
7234 17639 
```{r}
rldpvals=cbind(rld,valMI32, valSM32, valBOTH27, valBOTH32)
head(rldpvals)
dim(rldpvals)
table(complete.cases(rldpvals))
```
```{r}
write.csv(rldpvals, "Symbiont_RLDandPVALS.csv", quote=F)
```

```{r}
colnames(rld)=paste(colData$treat)
head(rld)
```

```{r}
library(RColorBrewer)
# Sample distance heatmap. Sample to sample. How similar are each sample
sampleDists <- as.matrix(dist(t(rld)))
library(gplots)
heatmap.2(as.matrix(sampleDists), key=F, trace="none",
          col=colorpanel(100, "black", "white"),
          margin=c(10, 10), main="Sample Distance Matrix")
```

### VENN Diagram to include both up and down regulated genes in common for PC02
Standard way to show number of DEs. Making series of genes that are up and down in each by pulling on res function
making p adjusted cutoff.
```{r}
library(VennDiagram)

MI32_up=row.names(resMI32[resMI32$padj<0.1 & !is.na(resMI32$padj) & resMI32$log2FoldChange>0,])
length(MI32_up) #2077
MI32_down=row.names(resMI32[resMI32$padj<0.1 & !is.na(resMI32$padj) & resMI32$log2FoldChange<0,])
length(MI32_down) #2163
SM32_up=row.names(resSM32[resSM32$padj<0.1 & !is.na(resSM32$padj) & resSM32$log2FoldChange>0,])
length(SM32_up) #2611
SM32_down=row.names(resSM32[resSM32$padj<0.1 & !is.na(resSM32$padj) & resSM32$log2FoldChange<0,])
length(SM32_down) #4084

BOTH32_up=row.names(resBOTH32[resMI32$padj<0.1 & !is.na(resBOTH32$padj) & resBOTH32$log2FoldChange>0,])
length(BOTH32_up) #1833
BOTH32_down=row.names(resBOTH32[resBOTH32$padj<0.1 & !is.na(resBOTH32$padj) & resBOTH32$log2FoldChange<0,])
length(BOTH32_down) #1018
BOTH27_up=row.names(resBOTH27[resBOTH27$padj<0.1 & !is.na(resBOTH27$padj) & resBOTH27$log2FoldChange>0,])
length(BOTH27_up) #2763
BOTH27_down=row.names(resBOTH27[resBOTH27$padj<0.1 & !is.na(resBOTH27$padj) & resBOTH27$log2FoldChange<0,])
length(BOTH27_down) #2395
```

overall number of genes with correct p value and no NA
```{r}
MI32=row.names(resMI32[resMI32$padj<0.1 & !is.na(resMI32$padj),])
SM32=row.names(resSM32[resSM32$padj<0.1 & !is.na(resSM32$padj),])
BOTH32=row.names(resBOTH32[resBOTH32$padj<0.1 & !is.na(resBOTH32$padj),])
BOTH27=row.names(resSM32[resBOTH27$padj<0.1 & !is.na(resBOTH27$padj),])
```

#### UP. Total number of up genes without repeated isogroups
```{r}
pdegs05_up=union(BOTH32_up, BOTH27_up)
length(pdegs05_up)
```

**4474**

```{r}
pdegs05_up=union(MI32_up, SM32_up)
length(pdegs05_up)
```
**4413**

#### DOWN. Total number of down genes without repeated isogroups
```{r}
pdegs05_down=union(BOTH32_down,BOTH27_down)
length(pdegs05_down)
```
**3296**

```{r}
pdegs05_down=union(MI32_down,SM32_down)
length(pdegs05_down)
```
**5953**

#### ALL
```{r}
pdegs05=union(MI32,SM32)
length(pdegs05)
```
**9296**

```{r}
pdegs05=union(BOTH32,BOTH27)
length(pdegs05)
```
**7746**

#### do UP, DOWN, ALL. Can change list to plot different things
```{r}
candidates=list("MI32"=MI32_up, "SM"=SM32_up)

prettyvenn=venn.diagram(
  x = candidates,
  filename=NULL,
  col = "transparent",
  fill = c("coral2", "forestgreen"),
  alpha = 0.5,
  # label.col = c("darkred", "white", "darkgreen", "white", "white", "white", "blue4"),
  cex = 2.5,
  fontfamily = "sans",
  fontface = "bold",
  cat.default.pos = "text",
  cat.col = c("darkred", "darkgreen"),
  cat.cex = 2.5,
  cat.fontfamily = "sans",
  cat.dist = c(0.08, 0.08),
  cat.pos = 1
);
grid.draw(prettyvenn)
```

```{r}
candidates=list("MI32"=MI32_down, "SM"=SM32_down)

prettyvenn=venn.diagram(
  x = candidates,
  filename=NULL,
  col = "transparent",
  fill = c("coral2", "forestgreen"),
  alpha = 0.5,
  # label.col = c("darkred", "white", "darkgreen", "white", "white", "white", "blue4"),
  cex = 2.5,
  fontfamily = "sans",
  fontface = "bold",
  cat.default.pos = "text",
  cat.col = c("darkred", "darkgreen"),
  cat.cex = 2.5,
  cat.fontfamily = "sans",
  cat.dist = c(0.08, 0.08),
  cat.pos = 1
);
grid.draw(prettyvenn)
```


```{r}
candidates=list("M"=MI32_up, "SM"=SM32_up)

prettyvenn=venn.diagram(
  x = candidates,
  filename=NULL,
  col = "transparent",
  fill = c("coral2", "forestgreen"),
  alpha = 0.5,
  # label.col = c("darkred", "white", "darkgreen", "white", "white", "white", "blue4"),
  cex = 2.5,
  fontfamily = "sans",
  fontface = "bold",
  cat.default.pos = "text",
  cat.col = c("darkred", "darkgreen"),
  cat.cex = 2.5,
  cat.fontfamily = "sans",
  cat.dist = c(0.08, 0.08),
  cat.pos = 1
);
grid.draw(prettyvenn)
```

### Heat map of sample distances for pco2
We only use columns 1 through 8 to cut off pvalues becase we don't want to visulise them
```{r}
setwd("C:/Users/Bruce Gan/Desktop/BI586/HW2")
rldpvals <- read.csv(file="Symbiont_RLDandPVALS.csv", row.names=1)
head(rldpvals)
rld=rldpvals[,1:8]
head(rld)
```

```{r}
sampleDists <- dist(t(rld))
sampleDistMatrix <- as.matrix( sampleDists )
treat=c( "MI_32", "MI_32", "MI_27", "MI_27", "SM_32", "SM_32", "SM_27", "SM_27")
colnames(sampleDistMatrix)=paste(treat)
rownames(sampleDistMatrix)=paste(treat)
```

Same heatmap as before but with color scales
```{r}
heat.colors = colorRampPalette(rev(c("blue","yellow","red")),bias=0.3)(100)
pheatmap(sampleDistMatrix,color = heat.colors,cex=0.9,border_color=NA,cluster_rows=T,cluster_cols=T)
```

###PCA looks at distance between samples. Focusing on PCA 1 and 2

head(pca)
```{r}
rld_t=t(rld)
pca <- prcomp(rld_t,center = TRUE, scale. = TRUE)
li <- pca$sdev^2 / sum(pca$sdev^2)
pc1v <- round(li[1] * 100, 1)
pc2v <- round(li[2] * 100, 1)
pca_s <- as.data.frame(pca$x)
head(pca_s)
pca_s <- pca_s[,c(1,2)]
pca_s$Samples = row.names(pca_s)
pca_s$treat=colData$treat
head(pca_s)
```

```{r}
cbPalette <- c("darkgoldenrod2",  "darkolivegreen3", "dodgerblue3", "red")
ggplot(pca_s, aes(PC1, PC2, color = treat, pch = treat)) +
  geom_point(size=3) +
  #  geom_text_repel(aes(label=Samples)) +
  scale_colour_manual(values=cbPalette)+
  theme_bw() +
  # geom_density2d(alpha=.5)+
  geom_polygon(alpha=.2)+
  xlab(paste0("PC1: ",pc1v,"% variance")) +
  ylab(paste0("PC2: ",pc2v,"% variance")) 
```
  
# tests if distances is significantly different. Treatment is having signficant effect 
```{r}
adonis(pca$x ~ treat, data = pca_s, method='eu', na.rm = TRUE)
```
                   Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)  
treat      3    129848   43283  3.9115 0.74578  0.013 *
Residuals  4     44263   11066         0.25422         
Total      7    174111                 1.00000   


#### Heatmaps for genes NS vs FR
```{r}
rldpvals <- read.csv(file="Symbiont_RLDandPVALS.csv", row.names=1)
head(rldpvals)
rld_site= rldpvals[,1:9]
head(rld_site)
gg=read.table("transcript2geneDescription.out.txt",sep="\t", row.names=1)
head(gg)
```


```{r}
nrow(rldpvals[rldpvals$padj.MI32<0.01& !is.na(rldpvals$padj.MI32),])
```
#1646

```{r}
nrow(rldpvals[rldpvals$padj.SM32<0.01& !is.na(rldpvals$padj.SM32),])
```
#3342

```{r}
nrow(rldpvals[rldpvals$padj.BOTH27<0.01& !is.na(rldpvals$padj.BOTH27),])
```
#2325

```{r}
nrow(rldpvals[rldpvals$padj.BOTH32<0.01& !is.na(rldpvals$padj.BOTH32),])
```
#1738

# I think this is the top 100 DEGs for each of the 4 treatments??

# MI32
```{r}
topnum= 100 # number of DEGS
head(rldpvals)
top100=head(rldpvals[order(rldpvals$padj.MI32), ],topnum)
head(top100)
length(top100[,1])
summary(top100)
###
library(pheatmap)
head(top100)
p.val=0.1 # FDR cutoff
conds=top100[top100$padj.MI32<=p.val & !is.na(top100$padj.MI32),]
length(conds[,1])

exp=conds[,1:8] # change numbers to be your vsd data columns
means=apply(exp,1,mean) # means of rows
explc=exp-means # subtracting them
head(explc)

ccol=colorRampPalette(rev(c("red","chocolate1","#FEE090","grey10", "cyan3","cyan")))(100)
col0=colorRampPalette(rev(c("chocolate1","#FEE090","grey10", "cyan3","cyan")))(100)

pheatmap(explc,cluster_cols=T,scale="row",color=col0, show_rownames = F)
```

#SM32
```{r}
topnum= 100 # number of DEGS
head(rldpvals)
top100=head(rldpvals[order(rldpvals$padj.SM32), ],topnum)
head(top100)
length(top100[,1])
summary(top100)
###
library(pheatmap)
head(top100)
p.val=0.1 # FDR cutoff
conds=top100[top100$padj.SM32<=p.val & !is.na(top100$padj.SM32),]
length(conds[,1])

exp=conds[,1:8] # change numbers to be your vsd data columns
means=apply(exp,1,mean) # means of rows
explc=exp-means # subtracting them
head(explc)

ccol=colorRampPalette(rev(c("red","chocolate1","#FEE090","grey10", "cyan3","cyan")))(100)
col0=colorRampPalette(rev(c("chocolate1","#FEE090","grey10", "cyan3","cyan")))(100)

pheatmap(explc,cluster_cols=T,scale="row",color=col0, show_rownames = F)
```

# BOTH27
```{r}
topnum= 100 # number of DEGS
head(rldpvals)
top100=head(rldpvals[order(rldpvals$padj.BOTH27), ],topnum)
head(top100)
length(top100[,1])
summary(top100)
###
library(pheatmap)
head(top100)
p.val=0.1 # FDR cutoff
conds=top100[top100$padj.BOTH27<=p.val & !is.na(top100$padj.BOTH27),]
length(conds[,1])

exp=conds[,1:8] # change numbers to be your vsd data columns
means=apply(exp,1,mean) # means of rows
explc=exp-means # subtracting them
head(explc)

ccol=colorRampPalette(rev(c("red","chocolate1","#FEE090","grey10", "cyan3","cyan")))(100)
col0=colorRampPalette(rev(c("chocolate1","#FEE090","grey10", "cyan3","cyan")))(100)

pheatmap(explc,cluster_cols=T,scale="row",color=col0, show_rownames = F)
```

# BOTH32
```{r}
topnum= 100 # number of DEGS
head(rldpvals)
top100=head(rldpvals[order(rldpvals$padj.BOTH32), ],topnum)
head(top100)
length(top100[,1])
summary(top100)
###
library(pheatmap)
head(top100)
p.val=0.1 # FDR cutoff
conds=top100[top100$padj.BOTH32<=p.val & !is.na(top100$padj.BOTH32),]
length(conds[,1])

exp=conds[,1:8] # change numbers to be your vsd data columns
means=apply(exp,1,mean) # means of rows
explc=exp-means # subtracting them
head(explc)

ccol=colorRampPalette(rev(c("red","chocolate1","#FEE090","grey10", "cyan3","cyan")))(100)
col0=colorRampPalette(rev(c("chocolate1","#FEE090","grey10", "cyan3","cyan")))(100)

pheatmap(explc,cluster_cols=T,scale="row",color=col0, show_rownames = F)
```




###################################Heatmap for the genes in common

#Not sure how to set this up but here goes...

# MI32 & SM32
```{r}
rldpvals <- read.csv(file="Symbiont_RLDandPVALS.csv", row.names=1)
head(rldpvals)
p.val=0.1 # FDR cutoff
conds=rldpvals[rldpvals$padj.MI32<=p.val & !is.na(rldpvals$padj.MI32) & rldpvals$padj.SM32<=p.val & !is.na(rldpvals$padj.SM32),]
rld_data= conds[,c(1:8)]
head(rld_data)
nrow(rld_data)
gg=read.table("transcript2geneDescription.out.txt",sep="\t", row.names=1)
library(pheatmap)
means=apply(rld_data,1,mean) # means of rows
explc=rld_data-means # subtracting them

ccol=colorRampPalette(rev(c("red","chocolate1","#FEE090","grey10", "cyan3","cyan")))(100)
col0=colorRampPalette(rev(c("chocolate1","#FEE090","grey10", "cyan3","cyan")))(100)

pheatmap(explc,cluster_cols=T,scale="row",color=col0, show_rownames = F)
```

# BOTH27 and BOTH32
```{r}
rldpvals <- read.csv(file="Symbiont_RLDandPVALS.csv", row.names=1)
head(rldpvals)
p.val=0.1 # FDR cutoff
conds=rldpvals[rldpvals$padj.BOTH27<=p.val & !is.na(rldpvals$padj.BOTH27) & rldpvals$padj.BOTH32<=p.val & !is.na(rldpvals$padj.BOTH32),]
rld_data= conds[,c(1:8)]
head(rld_data)
nrow(rld_data)
gg=read.table("transcript2geneDescription.out.txt",sep="\t", row.names=1)
library(pheatmap)
means=apply(rld_data,1,mean) # means of rows
explc=rld_data-means # subtracting them

ccol=colorRampPalette(rev(c("red","chocolate1","#FEE090","grey10", "cyan3","cyan")))(100)
col0=colorRampPalette(rev(c("chocolate1","#FEE090","grey10", "cyan3","cyan")))(100)

pheatmap(explc,cluster_cols=T,scale="row",color=col0, show_rownames = F)
```

## This is where I got stuck :(

# Make annotation table for pheatmap
```{r}
ann = data.frame(cond = c('MI_32', 'MI_32.1', 'MI_27', 'MI_27.1', 'SM_32', 'SM_32.1', 'SM_27', 'SM_27.1'))
rownames(ann) <- names(explc)
```

ann = data.frame(cond = c('MI_32', 'MI_32.1', 'MI_27', 'MI_27.1', 'SM_32', 'SM_32.1', 'SM_27', 'SM_27.1'))
rownames(ann) <- names(explc)

# Set colors
```{r}
Var1        <- c("darkgoldenrod2", "darkolivegreen3", "dodgerblue3", "cyan", "aliceblue", "azure", "blue", "brown")
names(Var1) <- c('MI_32', 'MI_32.1', 'MI_27', 'MI_27.1', 'SM_32', 'SM_32.1', 'SM_27', 'SM_27.1')
anno_colors <- list(cond = Var1)
```

Var1        <- c("darkgoldenrod2",  "darkolivegreen3", "dodgerblue3")
names(Var1) <- c("7.5", "7.6", "8")
anno_colors <- list(cond = Var1)

```{r}
pheatmap(as.matrix(explc),annotation_col=ann,annotation_colors=anno_colors,cex=1.2,color=col0,border_color=NA,clustering_distance_rows="correlation",clustering_distance_cols="correlation", show_rownames=T)
```


pheatmap(as.matrix(explc),annotation_col=ann,annotation_colors=anno_colors,cex=1.2,color=col0,border_color=NA,clustering_distance_rows="correlation",clustering_distance_cols="correlation", show_rownames=T)



```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
